common/src/main/java/com/wynntils/screens/overlays/selection/CustomBarSelectionScreen.java

```diff
@@ -0,0 +1,242 @@
+/*
+ * Copyright © Wynntils 2024.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.screens.overlays.selection;
+
+import com.mojang.blaze3d.vertex.PoseStack;
+import com.wynntils.core.components.Managers;
+import com.wynntils.core.consumers.features.Feature;
+import com.wynntils.core.consumers.screens.WynntilsScreen;
+import com.wynntils.core.persisted.config.OverlayGroupHolder;
+import com.wynntils.features.overlays.CustomBarsOverlayFeature;
+import com.wynntils.utils.mc.McUtils;
+import com.wynntils.utils.render.Texture;
+import com.wynntils.utils.render.buffered.BufferedRenderUtils;
+import com.wynntils.utils.render.type.BarTexture;
+import com.wynntils.utils.render.type.HealthTexture;
+import com.wynntils.utils.render.type.ManaTexture;
+import com.wynntils.utils.render.type.ObjectivesTextures;
+import com.wynntils.utils.render.type.UniversalTexture;
+import com.wynntils.utils.type.Pair;
+import java.util.List;
+import net.minecraft.ChatFormatting;
+import net.minecraft.client.gui.GuiGraphics;
+import net.minecraft.client.gui.components.Button;
+import net.minecraft.client.gui.components.Tooltip;
+import net.minecraft.client.gui.components.events.GuiEventListener;
+import net.minecraft.client.gui.screens.Screen;
+import net.minecraft.network.chat.Component;
+import net.minecraft.sounds.SoundEvents;
+import org.lwjgl.glfw.GLFW;
+
+public class CustomBarSelectionScreen extends WynntilsScreen {
+    // The order of these matters, they must match that in CustomBarsOverlayFeature
```

Comment:
Why is that?

Reply:
Because when the bar is actually added, it needs the index of it in the list of OverlayGroupHolder's so it uses `barTypeIndex` to track which bar type you currently have selected so that the same index can be used when getting the group for the specific bar type

Reply:
You could use the class of the overlay, which the holder provides from a getter. That way, you can save the overlay type in this variable, and we won't break it ever by forgetting to update this.

Reply:
I would like to think we won't forget to update this as if any new bar types are ever added they will probably use this screen to add it and will notice it not working then

Reply:
Okay, let's skip this for now.

Reply:
Even if we add a new type, it probably won't cause issues with the indexing, as it'll be added after the last holder.

Reply:
I've added a comment to the feature just in case